src/condor_utils/file_transfer.cpp

```diff
@@ -4718,6 +4726,47 @@ MyString FileTransfer::GetSupportedMethods() {
 	return method_list;
 }
 
+int FileTransfer::InitializeJobPlugins(const ClassAd &job, CondorError &e, StringList &infiles)
+{
+	if ( ! I_support_filetransfer_plugins || ! plugin_table) {
+		return 0;
+	}
+
+	std::string job_plugins;
+	if ( ! job.LookupString(ATTR_TRANSFER_PLUGINS, job_plugins)) {
+		return 0;
+	}
+
+	StringTokenIterator plugins(job_plugins, 100, ";");
+	for (const char * plug = plugins.first(); plug != NULL; plug = plugins.next()) {
+		const char * colon = strchr(plug, ':');
+		if (colon) {
+			MyString methods; methods.set(plug, colon - plug);
+
+			// add the plugin to the front of the input files list
+			MyString plugin_path(colon + 1);
+			plugin_path.trim();
+			if (! infiles.file_contains(plugin_path.c_str())) {
+				infiles.insert(plugin_path.c_str());
+			}
+			// use the file basename as the plugin name, so that when we invoke it
+			// we will invoke the copy in the input sandbox
+			MyString plugin(condor_basename(plugin_path.c_str()));
+
+			InsertPluginMappings(methods, plugin);
```

Comment:
Can this be an opportunity to switch `InsertPluginMethods` to accept `std::string` instead?

At least above, neither `plugin` nor `methods` utilize any `MyString` magic.  Admittedly, `plugin_path` does -- although I'm not entirely sure `trim` isn't just repeating work `StringTokenIterator ` is also doing.

Reply:
The plugin tables all need to be refactored because they have icky parallel simple maps of MyStrings, so when we refactor those would be a good time to go std::string.  Also we want the tags to be case-insensitive.